### DISCUSSION TITLE
Laravel Sanctum

### DISCUSSION THREAD

**Discussion by patie** [adonisjs/core#2271] [Ideas]
Hi, 

is there something like Laravel Sanctum? https://laravel.com/docs/8.x/sanctum

**Comment by McSneaky** [comment]
Not entierly sure what it provides. But in AdonisJS you when setting up auth you can check that you want `web` and `api` based authentication https://preview.adonisjs.com/guides/auth/setup

`web` uses the cookie (which you can use for SPAs) https://preview.adonisjs.com/guides/auth/web-guard
`api` uses API tokens https://preview.adonisjs.com/guides/auth/api-guard

It seems to provide the same functionality

**Reply by patie** [reply]
yep, i think its same, thaank you!

**Comment by begueradj** [comment]
The main advantage of Sanctum is related to SPA authentication without the need to use API tokens. From this perspective, there is nothing similar available (yet) for AdonisJs.

**Reply by RomainLanz** [reply]
You can use the `web` guard for that.
